File: app/src/main/java/bassamalim/hidaya/activities/WelcomeActivity.java
package bassamalim.hidaya.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import bassamalim.hidaya.R;
import bassamalim.hidaya.databinding.ActivityWelcomeBinding;
import bassamalim.hidaya.other.Utils;

public class WelcomeActivity extends AppCompatActivity {

    private ActivityWelcomeBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Utils.onActivityCreateSetTheme(this);
        Utils.setLocale(this, null);
        binding = ActivityWelcomeBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        if (savedInstanceState == null)
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.settings, new Settings.SettingsFragment()).commit();

        setListeners();
    }

    private void setListeners() {
        binding.saveBtn.setOnClickListener(v -> {
            binding.settings.setVisibility(View.GONE);
            binding.saveBtn.setVisibility(View.GONE);
            binding.disclaimerSpace.setVisibility(View.VISIBLE);
        });

        binding.agreed.setOnClickListener(view -> {
            Intent intent = new Intent(this, Splash.class);
            startActivity(intent);
            finish();
        });
        binding.rejected.setOnClickListener(view -> launchAnyWay());
    }

    private void launchAnyWay() {
        Intent intent = new Intent(this, MainActivity.class);
        intent.putExtra("located", false);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/bassamalim/hidaya/activities/WelcomeActivity.kt
package bassamalim.hidaya.activities

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.preference.PreferenceManager
import bassamalim.hidaya.R
import bassamalim.hidaya.ui.components.MyButton
import bassamalim.hidaya.ui.components.MyText
import bassamalim.hidaya.ui.theme.AppTheme
import bassamalim.hidaya.utils.ActivityUtils

class WelcomeActivity: AppCompatActivity() {

    private lateinit var pref: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActivityUtils.myOnActivityCreated(this)

        pref = PreferenceManager.getDefaultSharedPreferences(this)

        setContent {
            AppTheme {
                UI()
            }
        }
    }

    @Composable
    private fun UI() {
        Box(
            Modifier.background(AppTheme.colors.background)
        ) {
            Column(
                Modifier
                    .fillMaxSize()
                    .padding(6.dp),
                verticalArrangement = Arrangement.SpaceEvenly,
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                MyText(
                    text = stringResource(R.string.welcome_message),
                    fontSize = 26.sp
                )

                Settings.AppearanceSettings(this@WelcomeActivity, pref)

                MyButton(
                    text = stringResource(R.string.save),
                    fontSize = 24.sp,
                    innerPadding = PaddingValues(vertical = 2.dp, horizontal = 25.dp),
                    modifier = Modifier.padding(bottom = 10.dp)
                ) {
                    val intent = Intent(
                        this@WelcomeActivity, LocationActivity::class.java
                    )
                    intent.action = "initial"
                    startActivity(intent)

                    pref.edit()
                        .putBoolean("new_user", false)
                        .apply()

                    finish()
                }
            }
        }
    }

}